zebra-chain/src/block.rs

```diff
@@ -46,9 +46,9 @@ pub use arbitrary::LedgerState;
 #[cfg_attr(any(test, feature = "proptest-impl"), derive(Serialize))]
 pub struct Block {
     /// The block header, containing block metadata.
-    pub header: Header,
+    pub header: Arc<Header>,
```

Comment:
Using `Arc<Header>` reduces memory copies and memory usage, but it needs updates to a bunch of tests.